src/app/webdav/cloud-file-service.service.spec.ts

```diff
@@ -1,23 +1,72 @@
+import { User } from 'app/user/user';
+import { AppConfig } from 'app/app-config/app-config';
+import webdav from 'webdav';
 
 describe('CloudFileService', () => {
-  beforeEach(() => TestBed.configureTestingModule({}));
+  let cloudFileService: CloudFileService;
+  let sessionService: jasmine.SpyObj<SessionService>;
+  let sessionSpy;
+  let clientSpy;
 
   beforeEach(() => {
-    TestBed.configureTestingModule({
-      providers: [
-        SessionService
-      ],
-    })
-    .compileComponents();
-  });
-
-  // TODO: write tests!
-  // it('should be created', () => {
-  //   const service: CloudFileService = TestBed.get(CloudFileService);
-  //   expect(service).toBeTruthy();
-  // });
+    AppConfig.settings = {
+      site_name: '',
+      database: {name: 'unit-tests', remote_url: '', timeout: 60000, outdated_threshold_days: 0, useTemporaryDatabase: true},
+      webdav: {remote_url: 'test-url'}
+    };
+
+    sessionSpy = jasmine.createSpyObj('SessionService', ['getCurrentUser']);
+    clientSpy = jasmine.createSpyObj('client', ['getDirectoryContents', 'createDirectory', 'getFileContents', 'putFileContents']);
+
+    TestBed.configureTestingModule({ providers: [
+      CloudFileService,
+      { provide: SessionService, useValue: sessionSpy}]
+    });
+
+    cloudFileService = TestBed.get(CloudFileService);
+    cloudFileService['client'] = clientSpy;
+    sessionService = TestBed.get(SessionService);
+  });
+
+  it('.connect() should call sessionService.getCurrentUser()', () => {
+    cloudFileService.connect('user', 'pass');
+    expect(sessionService.getCurrentUser).toHaveBeenCalled();
+  });
+
+  it('.connect() should call webdav.createClient()', () => {
+    spyOn(webdav, 'createClient');
+    sessionService.getCurrentUser.and.returnValue(new User('user'));
+    cloudFileService.connect('user', 'pass');
+    expect(webdav.createClient).toHaveBeenCalledWith('test-url', {username: 'user', password: 'pass'});
+  });
+
+  it('.getDir() should call webdav.getDirectoryContents()', () => {
+    cloudFileService.getDir('testDir');
+    expect(clientSpy.getDirectoryContents).toHaveBeenCalledWith('testDir');
+  });
+
+  it('should create dir', () => {
+    cloudFileService.createDir('testDir');
+    expect(clientSpy.createDirectory).toHaveBeenCalledWith('testDir');
+  });
+
+  it('should check file existance', async() => {
+    spyOn(cloudFileService, 'getDir').and.returnValue(new Promise((resolve, reject) => {resolve('"basename": "filename"'); }));
+    expect(await cloudFileService.doesFileExist('filename')).toBe(true);
+    expect(await cloudFileService.doesFileExist('nonexistant')).toBe(false);
+  });
+
+  it('should get images', async() => {
+    spyOn(cloudFileService, 'doesFileExist').and.returnValue(new Promise((resolve, reject) => {resolve(true); }));
+    await cloudFileService.getImage('filepath');
+    expect(clientSpy.getFileContents).toHaveBeenCalledWith('filepath');
+  });
```

Comment:
you could add one more test for the case when there is no image and the default image should be returned